docs/TemplateManagementCLI.md

```diff
@@ -0,0 +1,101 @@
+```
+* Oras Login
+
+The [oras](https://github.com/deislabs/oras) tool oras.exe is packed in our repo, users can directly use it for login as follows.
+
+```
+>.\oras.exe login <registry> -u <username> -p <password>
+```
+
+## Push
+To push a template collection, the command is: 
+
+```
+push <ImageReference> InputTemplateFolder [ -n | --NewBaseLayer]
+```
+| Value | index |Optionality |  Description |
+| ----- | ----- | ----- |----- |
+| ImageReference |0| Required |  Image reference: \<Registry>\/\<Name> \[:\<Tag>]  (Image name should be lowcase.)|
+|InputTemplateFolder | 1 |Required |Input template folder. |
+
+| Option | Name | Optionality | Default | Description |
+| ----- | ----- | ----- |----- |----- |
+| -n | BuildNewBaseLayer | Optional | false | Ignore previous base layer and build new layer. |
+
+Example usage to push a collection of templates to ACR image from a folder:
+
+```
+>.\Microsoft.Health.Fhir.Liquid.Converter.Tool.exe push testacr.azurecr.io/templatetest:default myInputFolder
+```
+
+When pushing templates, all files except files in hidden image folder ("./.image/") will be packed as new template image. If the folder is unpacked from a previous template image, our tool will pack all user modified files into the user layer and then push all layers to ACR (The base layer is stores in hidden folder "./.image/"). If customers using -n as parameter, all templates will be packed together and be pushed as one layer to ACR.
+
+After successfully pushing an image, relevant information including layers' digests and image digest will output to users. Here is an output example, users should remember the image digest which exactly index an image:
+
+```
+Uploading 4085e9f97630 layer2.tar.gz
+Uploading 4157f847ecb1 layer1.tar.gz
+Pushed testacr.azurecr.io/templatetest:default
+Digest: sha256:412ea84f1bb1a9d98345efb7b427ba89616ec29ac332d543eff9a2161ca12a58
+```
+
+## Pull 
+For pull operation, the command is 
+
+```
+pull <ImageReference> <OutputTemplateFolder> [ -f | --ForceOverride]
+```
+
+| Value | index |Optionality |  Description |
+| ----- | ----- | ----- |----- |
+| ImageReference |0| Required |  Image reference: \<Registry>\/\<Name>\[ @\<Digest> \| :\<Tag> \] |
+|OutputTemplateFolder | 1 |Required | Output template folder. |
+
+| Option | Name | Optionality | Default | Description |
+| ----- | ----- | ----- |----- |----- |
+| -f | ForceOverride | Optional | false | Force to override the output folder. |
+
+Example usage to pull an image of templates in a folder:
+
+```
+>.\Microsoft.Health.Fhir.Liquid.Converter.Tool.exe pull testacr.azurecr.io/templatetest@sha256:412ea84f1bb1a9d98345efb7b427ba89616ec29ac332d543eff9a2161ca12a58 myOutputFolder
+```
+
+When pulling a collection of templates, a hidden folder "./.image/" will be created in output folder as well, which stores meta and layers' information. Users shouldn't modify this hidden folder which may lead to unexpected result.
```

Comment:
After a collection of templates is pulled, a hidden folder ".image/" which contains information of metadata and layers is also created in the output folder.